### ISSUE TITLE
config load issue after update to 2.1.0

### ISSUE THREAD

**Issue by andboss** [jimporter/mike#217]
# Summary

Configuration for building with-pdf is not working. I cannot provide the whole config, but need support how to log the issue line / affected config key to investigate further.

error appears:
```
error: join() argument must be str, bytes, or os.PathLike object, not 'RoundTrippableTag'
```

# Configuration and Logs

```
# base configuration for the page
INHERIT: !ENV [MKDOCS_FILE, 'mkdocs.yml']

plugins:
  # https://github.com/orzih/mkdocs-with-pdf
  with-pdf:
    enabled_if_env: ENABLE_PDF_EXPORT
    # verbose: true
    cover_subtitle: !ENV [PDF_COVER_SUBTITLE, SITE_DESCRIPTION, CI_PROJECT_ROOT_NAMESPACE, 'reusable-ci']
    # cover_logo: 'assets/images/logo.svg'
    toc_level: 3
    back_cover: true
    output_path: !ENV [PDF_OUTPUT_PATH, 'pdf/reusable-ci_use-mkdocs.pdf']
    custom_template_path: 'pdf-templates'
    # exclude pages for PDF rendering only, for global excludes see mkdocs-nav.yml
    #exclude_pages:
    #  - 'Appendix/*'
```

* Platform: GitLab 
* `mike` version: `2.1.0`

# Steps to Reproduce
1. call mike: `poetry run mike deploy --config-file $CONFIG_FILE --deploy-prefix $SITE_DIR --branch $PAGES_BRANCH $PAGES_DEPLOY_NAME`

# Expected Behavior

It should build like version `2.0.0`

**Comment by jimporter** [comment]
This works fine for me. You can get backtraces via `mike --debug <rest of args>` though.

**Comment by jimporter** [comment]
This should be fixed in 3351d5f. Could you test it out? If it looks ok, I'll tag a new version.

**Comment by andboss** [comment]
@jimporter i have no access to my debug workspace now, but from code review especially the tests it looks good 👍 

If there is a package for download or tagged version available i can try soon.

Thank You!

**Comment by jimporter** [comment]
@andboss Since it looks like you're using Poetry, you should be able to point the `mike` dependency to this Git repository: https://python-poetry.org/docs/dependency-specification/#git-dependencies

I think that would be equivalent from Poetry's POV to having a PyPI-published version.

**Comment by andboss** [comment]
... using this version:
```
    [tool.poetry.dependencies.mike]
    git = "https://github.com/jimporter/mike.git"
    rev = "3351d5f"
```

It works now 👍 🥳 

THANKS!

**Comment by jimporter** [comment]
Thanks for checking. I've merged that commit and released v2.1.1 with that change.
